sham-lib/sham/ir/builder.rkt

```diff
@@ -295,23 +298,28 @@
                        [ref (LLVMAddFunction llvm-module s fn-type)])
                   (hash-set! intrinsic-map s ref)
                   ref)))
-          (LLVMBuildCall builder ref rand-values (llvm-lhs s))]
+          (LLVMBuildCall builder ref rand-values (lhs ret-type s))]
          [(sham:ast:rator:external md lib-id id ret-type var-arg?)
           #:when (hash-has-key? ffi-mappings id)
-          (LLVMBuildCall builder (cdr (hash-ref ffi-mappings id)) rand-values "e")]
+          (define s (symbol->string id))
+          (define fn-value (cdr (hash-ref ffi-mappings id)))
+          (LLVMBuildCall builder fn-value rand-values (lhs ret-type s))]
          [(sham:ast:rator:external md lib-id id ret-type var-arg?)
           (define s (symbol->string id))
           (define fn-type (LLVMFunctionType (build-llvm-type ret-type env)
                                             (map LLVMTypeOf rand-values) var-arg?))
           (define fn-value (LLVMAddFunction llvm-module s fn-type))
           (add-ffi-mapping! id (cons lib-id fn-value))
-          (LLVMBuildCall builder fn-value rand-values (llvm-lhs s))]
+          (LLVMBuildCall builder fn-value rand-values (lhs ret-type s))]
          [(sham:ast:rator:racket md id rkt-fun type)
           (define s (symbol->string id))
           (define ct (compile-type type env))
           (define fn-type (internal-type-llvm ct))
           (define fn-value (LLVMAddFunction llvm-module s fn-type))
           (add-rkt-mapping! s (list rkt-fun (internal-type-racket ct) fn-value))
+          ;; Can a racket function ever return void?
+          ;; If so how do we check for a void return type?
+          ;; - Andre
           (LLVMBuildCall builder fn-value rand-values (llvm-lhs s))]
```

Comment:
the `type` in the pattern matching for this case should be a function type as well, I will add a check for void here using that as well.